include/s3select_oper.h

```diff
@@ -1312,36 +1313,50 @@ class arithmetic_operand : public base_statement
 
   virtual value& eval()
   {
```

Comment:
the code repeat itself
the following could save some code lines.
value lval = l->eval();
value rval = r->eval();
if ( lval->is_null() || rval->is_null()) { ..... ; return .... }

the rest no change (almost)

Reply:
**adding correction to my previous comment**
its not efficient to evaluate both sides (left and right)
the left should first evaluated (if false return false), if not the right side should evaluated